artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageReader.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.core.paging.cursor.impl;
+
+import org.apache.activemq.artemis.core.paging.PagedMessage;
+import org.apache.activemq.artemis.core.paging.cursor.PageCache;
+import org.apache.activemq.artemis.core.paging.cursor.PagePosition;
+import org.apache.activemq.artemis.core.paging.impl.Page;
+import org.jboss.logging.Logger;
+
+public class PageReader implements PageCache {
+   private static final Logger logger = Logger.getLogger(PageReader.class);
+
+   private Page page;
+
+   public PageReader(Page page) {
+      this.page = page;
+   }
+
+   @Override
+   public long getPageId() {
+      return page.getPageId();
+   }
+
+   @Override
+   public int getNumberOfMessages() {
+      return page.getNumberOfMessages();
+   }
+
+   @Override
+   public void setMessages(PagedMessage[] messages) {
+   }
+
+   @Override
+   public PagedMessage[] getMessages() {
```

Comment:
would this not be dangerous? either we have code relying on this, and need to support it, or we should remove it from the interface/api so people dont code against it.